examples/core/Graphormer/train.py

```diff
@@ -428,7 +433,10 @@ def collate(self, samples):
                 max_num_nodes - num_nodes[i],
             )
             shortest_path = F.pad(shortest_path, p3d, "constant", -1)
+            # +1 to distinguish the padding value and the edge feature
```

Comment:
Padding for what? Why is there a need for padding?

Reply:
Padding for node feature. Since we need to align the number of nodes in a batch, those invalid nodes will be padded with zero node feature and edge feature. Should I provide more detailed note?

Reply:
I was referring to the logic below related to `edata = th.cat((edata, th.zeros(1, edata.shape[1]).to(edata.device)), dim=0`. From this, there is only one padded edge. What does it stand for?

Reply:
Such a padded edge stands for "edge" between all unreachable node pairs. Since the path returned by `shortest_dist` contains edge IDs along the shortest path with -1 padding (can also be regarded as edge ID and interpreted as the last edge), we append a zero padding to the edge feature and thus all padded edge IDs (-1) will refer to this edge.

Reply:
Super! It will be very helpful to mention that the extra vector is for all disconnected node pairs.

Reply:
Sorry, I was wrong. The extra vector just stands for padded edge IDs (not for disconnected node pairs) since the length of shortest path is different for different node pairs.

An [example](https://github.com/dmlc/dgl/blob/40dcc7152d930a8761b84d9ed2041a32fa06972c/python/dgl/transforms/functional.py#L3874) of returned path from `shortest_dist`.

Reply:
Got it. Thanks for the clarification. It will be very helpful to have a comment for this.